src/OrchardCore.Modules/OrchardCore.Deployment/Steps/CustomFileDeploymentSource.cs

```diff
@@ -2,15 +2,9 @@
 
 namespace OrchardCore.Deployment.Steps;
 
-public class CustomFileDeploymentSource : IDeploymentSource
+public class CustomFileDeploymentSource
+    : DeploymentSourceBase<CustomFileDeploymentStep>
 {
-    public Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
-    {
-        if (step is not CustomFileDeploymentStep customFile)
-        {
-            return Task.CompletedTask;
-        }
-
-        return result.FileBuilder.SetFileAsync(customFile.FileName, Encoding.UTF8.GetBytes(customFile.FileContent));
-    }
+    protected override async Task ProcessAsync(CustomFileDeploymentStep step, DeploymentPlanResult result)
+        => await result.FileBuilder.SetFileAsync(step.FileName, Encoding.UTF8.GetBytes(step.FileContent));
```

Comment:
```suggestion
    protected override Task ProcessAsync(CustomFileDeploymentStep step, DeploymentPlanResult result)
        => result.FileBuilder.SetFileAsync(step.FileName, Encoding.UTF8.GetBytes(step.FileContent));
```

Reply:
Why you didn't awaited?!!

Reply:
@hishamco  you don't need to await in this case. you can simply return delegate return of the task to the SetFuleAsync method. this is better optimization to the code. Unless you need to await due to logic in your body, you should not await.